### ISSUE TITLE
Quick Question on adam.py

### ISSUE THREAD

**Issue by btahir** [angetato/Optimizers-for-Tensorflow#2]
Awesome work! The adam.py implementation is very useful! I just had a quick question about:

`alpha_t =  tf.sqrt(1 - beta2_power) / (1 - beta1_power)`

What does this achieve? Wouldn't the original Adam implementation be?

`var - lr * m_t / (tf.sqrt(v_t) + eps)`

rather than 

`var - lr * (m_t*alpha_t) / (tf.sqrt(v_t) + eps)`

Are you adding weight decay with alpha_t maybe?

**Comment by angetato** [comment]
Hello @btahir,
I know my answer is late but for those with the same question here is the answer : 
In the original  [paper](https://arxiv.org/pdf/1412.6980.pdf), see algorithm 1, they use 
![equation](https://latex.codecogs.com/gif.latex?\widehat{m_t}&space;=&space;m_t&space;/(1-\beta_1&space;^{t}))
 instead of `m_t` (same for `v_t`)  to compute the update. Thus : 
![equation](https://latex.codecogs.com/gif.latex?var&space;-&space;lr&space;*&space;\hat{m_t}&space;/\sqrt{\hat{v_t}&space;&plus;&space;\varepsilon}&space;=&space;var&space;-&space;lr&space;*&space;(m_t*alpha_t)/\sqrt{v_t&space;&plus;&space;\varepsilon})
Thanks.
